cupy/core/fusion.pyx

```diff
@@ -815,8 +818,10 @@ class _FusionHistory(object):
                                      for s, t in zip(out_cvars, out_params))
 
             submodule_code += self._emit_premap_code(in_params, operation)
-            submodule_code += 'typedef {} type_in0_raw;\n'.format(
-                postmap_ctype)
+            use_cub = ACCELERATOR_CUB in _accelerator._reduction_accelerators
+            if not use_cub:
+                submodule_code += 'typedef {} type_in0_raw;\n'.format(
+                    postmap_ctype)
```

Comment:
In general CUB expects a different `type_in0_raw`, so without doing this there'd be a redeclaration error. I consider this an imperfect workaround, but it leads to the smallest amount of code change IMHO.